Update release process steps

aanm opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/10035)
<!-- Reviewable:end -->

aanm committed:
contributing/release: fix typo in stable release process

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
contributing/release: remove Update versions step

This step is automatically generated by a command ran afterwards:
`make -C install/kubernetes clean all`

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
contributing/release: add step to add release notes to CHANGELOG.md

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
contributing/release: change Github release step

We will include the automated release notes generated in the Github
release page.

Signed-off-by: André Martins <andre@cilium.io>

joestringer commented on Documentation/contributing/release/stable.rst:
```diff
@@ -92,6 +91,10 @@ If you intent to release a new feature release, see the
             [...]
             <<end of add-to-NEWS.rst>>
 
+#. Generate the release notes by running the instructions provided in github.com/cilium/release
```
This URL doesn't work, are you planning to push that repo before merging?

aanm replied:
No, but the repo will have that name ;-)

joestringer reviewed (approved):
Looks good. Won't backport cleanly to v1.5 because of docs refactoring. (Might not even backport to v1.6 for the same reason..)

aanm committed:
contributing/release: remove update NEWS.rst step

`git shortlog v1.0.2..` can be executed by anyone in their terminal and
since we now have auto-generated release-notes it does not make sense to
run this step anymore.

Signed-off-by: André Martins <andre@cilium.io>
